admin-front/src/components/TimeSeries/hooks/useGetTimeSeriesData.tsx

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useState } from 'react';
+import moment from 'moment';
+import useGetDataFromAPI from 'shared/hooks/useGetDataFromAPI';
+import { getTimeSeriesFilteredByOccurrence, getDateFormat } from '../functions';
+import {
+  Month,
+  Resolution,
+  TimeSeriesRecord,
+  TimeSeriesType,
+} from '../utility';
+
+interface TimeSeriesData {
+  timeStamps: string[];
+  timeStampValues: number[];
+  firstYearTimeSeriesRecord?: TimeSeriesRecord;
+  fetchedTimeSeries?: TimeSeriesRecord[];
+}
+
+const TimeSeriesPaths = {
+  nftCount: '/analytics/sales/nftCount/timeseries',
+  priceVolume: '/analytics/sales/priceVolume/timeseries',
+};
+
+interface UseGetTimeSeriesDataProps {
+  timeSeriesType: TimeSeriesType;
+  resolution: Resolution;
+  year: number;
+  month: Month;
+}
+
+const UseGetTimeSeriesData = ({
+  timeSeriesType,
+  resolution,
+  year,
+  month,
+}: UseGetTimeSeriesDataProps): TimeSeriesData => {
+  const [timeStamps, setTimeStamps] = useState<string[]>([]);
+  const [timeStampValues, setTimeStampValues] = useState<number[]>([]);
+
+  const queryStr = resolution ? `?resolution=${resolution}` : undefined;
+  const { data: fetchedTimeSeries } = useGetDataFromAPI<TimeSeriesRecord[]>({
+    path: TimeSeriesPaths[timeSeriesType],
+    queryStr,
+  });
+
+  useEffect(() => {
+    if (fetchedTimeSeries) {
+      const timeSeriesFilteredByOccurrence = getTimeSeriesFilteredByOccurrence({
+        timeSeries: fetchedTimeSeries,
+        occurrence: { year, month },
+      });
+
+      setTimeStamps(
+        timeSeriesFilteredByOccurrence.map((record: TimeSeriesRecord) => {
+          const dateFormat = getDateFormat({ resolution, month });
+          return moment.unix(record.timestamp).format(dateFormat);
+        }),
+      );
+
+      setTimeStampValues(
+        timeSeriesFilteredByOccurrence.map(
+          (record: TimeSeriesRecord) => record.value,
+        ),
+      );
+    }
+  }, [fetchedTimeSeries, year, month]);
+
+  return {
+    timeStamps,
+    timeStampValues,
+    firstYearTimeSeriesRecord: fetchedTimeSeries?.[0],
```

Comment:
Returning this field here feels a bit weird somehow. Seems to me to belong to the call-site of this `useGetTimeSeriesData` rather than to here because it is so specific, and it can be just as easily based on `fetchedTimeSeries` in the call-site (ie whoever is calling this code can just do this `fetchedTimeSEries?.[0]`).